### ISSUE TITLE
Pinia stopped working after updating Nuxt to 2.13 (and beyond)

### ISSUE THREAD

**Issue by spetex** [posva/pinia#206]
Hello,

First: thanks for the work put in this library, it really is a fresh breeze after using vuex for some time.

To the point:

I have just recently updated our project to new version of nuxt (v.2.13) and I have encountered a problem with pinia.
Here is the transcript of the error:

```
TypeError: Cannot read property 'control' of undefined
    at getInitialState (pinia.esm.js?d071:119)
    at useStore (pinia.esm.js?d071:257)
    at setup (default.vue?2e42:40)
    at mergedSetupFn (vue-composition-api.module.js?750b:418)
    at eval (vue-composition-api.module.js?750b:637)
    at activateCurrentInstance (vue-composition-api.module.js?750b:569)
    at initSetup (vue-composition-api.module.js?750b:636)
    at VueComponent.wrappedData (vue-composition-api.module.js?750b:623)
    at getData (vue.runtime.esm.js?2b0e:4748)
    at initData (vue.runtime.esm.js?2b0e:4705)
```

It seems like an issue in this function:

```
function getInitialState(id) {
    var provider = stateProviders.get(getActiveReq());
    return provider && provider()[id];
}
```

I am not sure, but it seems to me it could be somehow related to work done on integrating the composition-api guys just recently did in the nuxt project.

Any insights on how to make this work again?

Thanks!

**Comment by posva** [comment]
Can you provide a boiled down repro using nuxt-composition-api?

**Comment by christianechevarria** [comment]
Not sure if it's related to the Nuxt Composition API updates @spetex mentioned, but I'm experiencing the behavior in https://github.com/posva/pinia/issues/179 this is what my `package.json` and `nuxt.config.js` look like respectively:

```
// package.json
{
  "name": "nuxt",
  "version": "1.0.0",
  "description": "A description",
  "author": "Christian George Echevarria-Reina",
  "private": true,
  "scripts": {
    "dev": "nuxt",
    "build": "nuxt-ts build",
    "old_start": "nuxt-ts start",
    "start": "nuxt-ts",
    "export": "nuxt-ts export",
    "serve": "nuxt-ts serve",
    "generate": "nuxt generate",
    "lint": "eslint --ext .js,.vue --ignore-path .gitignore .",
    "test": "jest"
  },
  "dependencies": {
    "@nuxtjs/apollo": "^4.0.1-rc.1",
    "@nuxtjs/dotenv": "^1.4.0",
    "@nuxtjs/pwa": "^3.0.0-0",
    "@tailwindcss/ui": "^0.3.0",
    "apollo-cache-inmemory": "^1.6.6",
    "body-parser": "^1.19.0",
    "express": "^4.17.1",
    "graphql-tag": "^2.10.3",
    "nuxt": "latest",
    "nuxt-composition-api": "^0.10.6",
    "pinia": "0.0.7",
    "tailwindcss": "^1.4.6"
  },
  "devDependencies": {
    "@babel/runtime-corejs3": "^7.10.1",
    "@nuxt/types": "^2.13.3",
    "@nuxt/typescript-build": "^2.0.0",
    "@nuxtjs/eslint-config": "latest",
    "@nuxtjs/eslint-module": "latest",
    "@nuxtjs/tailwindcss": "~2.0.0",
    "@vue/test-utils": "latest",
    "babel-eslint": "latest",
    "babel-jest": "latest",
    "core-js": "^3.6.5",
    "eslint": "latest",
    "eslint-config-prettier": "latest",
    "eslint-plugin-nuxt": "latest",
    "eslint-plugin-prettier": "latest",
    "jest": "latest",
    "prettier": "latest",
    "vue-jest": "latest"
  }
}
```

and 

```
// nuxt.config.js

export default {
  mode: 'universal',
  /*
   ** Headers of the page
   */
  head: {
    title: process.env.npm_package_name || '',
    meta: [
      { charset: 'utf-8' },
      { name: 'viewport', content: 'width=device-width, initial-scale=1' },
      {
        hid: 'description',
        name: 'description',
        content: process.env.npm_package_description || '',
      },
    ],
    link: [
      { rel: 'icon', type: 'image/x-icon', href: '/favicon.ico' },
      { rel: 'stylesheet', href: 'https://rsms.me/inter/inter.css' },
    ],
  },
  /*
   ** Customize the progress-bar color
   */
  loading: { color: '#fff' },
  /*
   ** Global CSS
   */
  css: [],
  /*
   ** Plugins to load before mounting the App
   */
  plugins: ['~plugins/filters.js'],
  /*
   ** Nuxt.js dev-modules
   */
  buildModules: [
    // Doc: https://github.com/nuxt-community/eslint-module
    '@nuxtjs/eslint-module',
    // Doc: https://github.com/nuxt-community/nuxt-tailwindcss
    '@nuxtjs/tailwindcss',
    '@nuxt/typescript-build',
    'pinia/nuxt',
    'nuxt-composition-api',
  ],
  /*
   ** Nuxt.js modules
   */
  modules: [
    '@nuxtjs/pwa',
    // Doc: https://github.com/nuxt-community/dotenv-module
    '@nuxtjs/dotenv',
    '@nuxtjs/apollo',
  ],
  // Give apollo module options
  apollo: {
    cookieAttributes: {
      expires: 7, // optional, default: 7 (days)
    },
    includeNodeModules: true, // optional, default: false (this includes graphql-tag for node_modules folder)
    authenticationType: 'Bearer', // optional, default: 'Bearer'
    // optional
    errorHandler: '~/plugins/apollo-error-handler.js',
    // required
    clientConfigs: {
      default: '~/apollo/clientConfig.js',
    },
  },
  serverMiddleware: ['~/api/index'],
  /*
   ** Build configuration
   */
  build: {
    /*
     ** You can extend webpack config here
     */
    extend(config, ctx) {},

    babel: {
      configFile: './babel.config.js',
    },
  },
}
```

I'm going to try rolling back Nuxt to see if it would work in versions lower than 2.12.x and report back the results

**Comment by christianechevarria** [comment]
Update: tried rolling back to 2.12.2 and fixed this particular issue -- although I had another issue where Nuxt would just refresh non-stop, can't say with certainty that it was related to this

**Comment by eugenioclrc** [comment]
Could you post a woring demo with pinia an nuxt?

**Comment by posva** [comment]
Closing due to inactivity. Please open a new issue with a reference to this one if you can follow up with more information.

**Comment by MartinMalinda** [comment]
This seems to be another occurence of https://github.com/posva/pinia/issues/161. Previous check is not enough because at some point `context.nuxtState` started existing in spa mode as well.

**Comment by MartinMalinda** [comment]
It seems to be fixed in 0.1.0
